get_info.php

```diff
@@ -1 +1,159 @@
+                $query = handle_get();
+                break;
+
+            default:
+                throw new Exception("Invalid Request");
+                break;
+        }
+    } else {
+        $error = 'No Request Method Found';
+        throw new Exception($error);
+    }
+} catch (Exception $e) {
+    $result = set_http_response(400);
+    date_default_timezone_set('America/Toronto');
+    error_log(
+        date("Y-m-d h:i:sa") . " : " . $e->getMessage() . "\n",
+        3,
+        "errors.log"
+    );
+    print json_encode($result, JSON_PRETTY_PRINT);
+    exit();
+}
+
+// Attempt to execute sql command and print response in json format.
+// If a sql error occurs, HTTP status 500 is returned.
+try {
+    // connect to the mysql database
+    $conn = new PDO(
+        "mysql:host=$servername;dbname=$database",
+        $username,
+        $password
+    );
+    // set the PDO error mode to exception
+    $conn->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
+    $stmt = $conn->prepare($query);
+    $stmt->execute();
+
+    if ($method == "GET") {
+        //fetch all results
+        $fetched = $stmt->fetchAll(PDO::FETCH_ASSOC);
+        $result['TYPE'] = $_GET["what"] . "_package";
+        $result['DATA'] = $fetched;
+    } else {
+        $result = set_http_response(202);
+    }
+
+    echo json_encode($result);
+} catch (PDOException $e) {
+    $result = set_http_response(500);
+    date_default_timezone_set('America/Toronto');
+    error_log(
+        date("Y-m-d h:i:sa") . " : " . $e->getMessage() . "\n",
+        3,
+        "errors.log"
+    );
+    print json_encode($result, JSON_PRETTY_PRINT);
+    exit();
+}
+
+/**
+ * This function handles GET requests and returns a sql query that gets the
+ * required data for the what according to the API documentation.
+ *
+ * @throws InvalidPage The requested what does not exist
+ * @return sql query to the corresponding what parameter
+ */
+function handle_get()
+{
+    switch ($_GET["what"]) {
+        case "tas":
+            return set_parameters(true);
+            break;
+        case "questions":
+            return get_questions();
+            break;
+        case "course_pairings":
+            return set_parameters(false);
+            break;
+        case "user_info":
+            $include_photo = true;
+            if (isset($_GET["include_photo"])) {
+                if ($_GET["include_photo"] == 'false') {
+                    $include_photo = false;
+                }
+            }
+            return get_user_info($_GET["user_id"], $include_photo);
+            break;
+
+        default:
+            throw new Exception("InvalidPage");
+            break;
+    }
+}
+
+/**
+ * This functions sets the parameters and calls get_course_associations and
+ * returns the results.
+ *
+ * @param num The HTTP status code
+ * @return array containing the HTTP status of request
+ */
+function set_parameters($ta_list)
+{
+    $user_id = $_GET["user_id"];
```

Comment:
Stop using globals inside functions.

Reply:
I used a global variable inside that function to improve code reuse. There are 2 cases that call that function and if I want to remove the global variable, it would need to go inside the handle_get function.

Reply:
Keep global variables to the top scope and pass in any data you need as parameters. The top scope should be the only part of the code that cares about state. All the state should be normalized first, and then the data can be dispatched to the appropriate functions.

In general (though not in this project), we would be writing test cases for these individual functions. If you rely on global state, then it's very hard to test the functions. Further, they could not be used as module--reliance on globals actually prevents reuse.